app/gui/view/graph-editor/src/component/breadcrumbs/project_name.rs

```diff
@@ -214,6 +216,24 @@ impl ProjectNameModel {
         self.commit(name);
     }
 
+    /// Confirm the given name as the current project name if it's valid.
+    fn try_commit(&self, name: impl Str) -> Result<(), String> {
+        let name = name.into();
+        Self::validate(&name)
+            .map_err(|error| format!("The project couldn't be renamed. {error}"))?;
+        self.commit(name);
+        Ok(())
+    }
+
+    /// Check whether the given name is a valid project name.
+    fn validate(name: impl Str) -> Result<(), String> {
+        let parser = Parser::new();
+        match parser.parse_line_ast(name).map(|ast| ast.shape().clone()) {
+            Ok(ast::Shape::Cons(_)) => Ok(()),
+            _ => Err("The project name should be in Upper_Snake_Case.".to_owned()),
```

Comment:
Minor thing, but I'd change it to `The project name should use the 'Upper_Snake' case.` This explains that this is a convention case in plain English rather than encoding the "case" information in the name.